source/expressions.tex

```diff
@@ -4155,11 +4155,13 @@
 If the operand is a \grammarterm{qualified-id} naming a non-static or variant member \tcode{m}
 of some class \tcode{C} with type \tcode{T}, the result has type ``pointer to member
 of class \tcode{C} of type \tcode{T}'' and is a prvalue designating \tcode{C::m}.
-Otherwise, if the type of the expression is \tcode{T}, the result has type ``pointer to
-\tcode{T}'' and is a prvalue that is the address of the designated object\iref{intro.memory}
-or a pointer to the designated function. \begin{note} In particular, the address of an
-object of type ``\cv{}~\tcode{T}'' is ``pointer to \cv{}~\tcode{T}'', with the same
-cv-qualification. \end{note}
+Otherwise, if the type of the expression is \tcode{T},
```

Comment:
It's unclear what expression "the expression" refers to here. I think it means "the operand when the operand is an lvalue", which leaves the behavior undefined when the operand is a *qualified-id* which does *not* name "a non-static or variant member `m` of some class `C` with type `T`."

I think we want to replace the sentence on lines 4155-4157 with: 
> If the operand is a *qualified-id*, it shall name a non-static or variant member `m` of some class `C` with type `T`; the result has type “pointer to member of class `C` of type `T`” and is a prvalue designating `C::m`.

And change later occurrences of "the expression" to "the operand", which must be an lvalue in those later occurrences.

Reply:
No, Casey: a *qualified-id* could be an lvalue *id-expression*. 4155-4157 is fine The *new* sentence needs to exclude the "*qualified-id* that is not an lvalue" case.

Reply:
I think that's an independent issue. The "Otherwise...expression" phrasing was already there and is used unchanged. I do agree that the preceding sentence is bad because it incorrectly mixes conditions and "shall" requirements. Can we fix that separately, please?

Reply:
Created #2919 to handle the bad if/otherwise ladder.